### PR TITLE
feat: Panels associated by id with specific connections

### LINKED ISSUE #90: Panels should be associated with specific servers
Currently there is a panel registry that caches instances of output panels by name. In cases where multiple servers are used, this does not allow for the same variable name to be used on multiple servers, as they will overwrite each other. We should map panel registry by name + serverurl.

### PR DESCRIPTION
**Pull request by bmingles**
Replaced panel registry with new PanelService
- Panels are now associated with connections by variable id (they were mapped by title before across all connections)
- Should be able to re-use the same variable name in different connections

**Testing**
- Start 2 servers and create a connection to both of them via the status bar connection picker (you also have the option of starting 2 pip servers which should auto create connections). This requires a venv setup with deephaven-server installed
- Run a script that produces a panel on the first connection
- Drag the script to the other connection and run it there
- Should see 2 panels with same title

resolves #90

### CODE REVIEW

File: src/services/DhService.ts [hunk 1]
```diff
@@ -363,18 +354,19 @@ export abstract class DhService<TDH = unknown, TClient = unknown>
...
-            .get(title)!
-            .webview.postMessage.bind(
-              this.panelRegistry.getOrThrow(title).webview
-            )
-        );
-      });
+      // TODO: The postMessage apis will be needed for auth in DHE (vscode-deephaven/issues/76).
+      // Leaving this here commented out for reference, but there are some issues
+      // to work through:
+      // 1. This seems to subscribe multiple times. Should see if can move it
```
Comment (mofojed) [inline src/services/DhService.ts:360]: Yes I think this should definitely be in the creation block above, or you need to dispose of the listener before this gets added again.

File: src/services/PanelService.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  VariableID,
+  VariablePanelMap,
+} from '../types';
+
+export class PanelService implements IPanelService, Disposable {
+  constructor() {
+    this._cnPanelMap = new URLMap<VariablePanelMap>();
+  }
+
+  private readonly _cnPanelMap: URLMap<VariablePanelMap>;
```
Comment (mofojed) [inline src/services/PanelService.ts:15]: `_cnPanelMap`? What's the `_cn` about?

File: src/services/DhService.ts [hunk 3]
```diff
@@ -363,18 +354,19 @@ export abstract class DhService<TDH = unknown, TClient = unknown>
...
+      // inside of the panel creation block
+      // 2. Not 100% sure we need to retrieve from the panel service instead of
```
Comment (mofojed) [inline src/services/DhService.ts:362]: Should just be able to re-use `panel`.

File: src/services/PanelService.ts (hunk 2 above)
Comment (bmingles) [reply src/services/PanelService.ts:15]: @mofojed "_cn" == "connection"

File: src/services/DhService.ts (hunk 1 above)
Comment (bmingles) [reply src/services/DhService.ts]: I updated these comments to be more decisive of what needs to be done.

**Review by mofojed — APPROVED** [review]

**Pull request merged**